DOC: expand docstring of exponweib distribution

pierre-haessig opened the pull request:
Hi, this is a small expansion to the docstring of the exponweib distribution. Indeed, it seems for 2 SO questions [1][2] that the meaning of the shape parameters is not obvious and doesn't match some classical definitions.

Also, it may be useful to add a "see also" section in `exponweib` pointing to `weibull_min` which seems to corresponds to the usual Weibull distribution, however its docstring doesn't say it explicitely (in fact, I find no references on the web for weibull_min and weibull_max except in scipy!). What do you think?

[1] https://stackoverflow.com/questions/17481672/fitting-a-weibull-distribution-using-scipy
[2] https://stackoverflow.com/questions/38287971/scipy-how-to-fit-weibull-distribution

rgommers commented on scipy/stats/_continuous_distns.py:
```diff
@@ -1446,14 +1446,32 @@ class exponweib_gen(rv_continuous):
+      (non-exponentiated) Weibull distribution.
+    * :math:`c` is the shape parameter of the usual Weibull law
+      (often named :math:`k`, but named :math:`a` in `numpy.random.weibull`).
+      Special values are :math:`c=1` and :math:`c=2` where Weibull distribution
```
this seems to be true only if `a=1`. So belongs in the `weibull_min` docstring rather than here?

pierre-haessig replied:
agreed

pierre-haessig commented:
Sorry for the long time lag. I've done the rebase and added a second commit with @rgommers suggestions. This impacts the docstring of `weibull_min` (for the better I hope).

Also, I'm linking here @WarrenWeckesser detailed feedback on the scipy-dev mailing list: https://mail.python.org/pipermail/scipy-dev/2019-January/023298.html. He explained clearly the origin of min and max suffixes in extreme value theory.

 Also, Warren disagree with my saying that `weibull_min` is "the usual Weibull distribution" since in two classical textbooks of extreme value theory it is in fact `weibull_max`. Still, I think that outside this field there is no ambiguity (and also on Wikipedia page on extreme value theory). I hope that the sentence I added in `weibull_min` docstring is well balanced.

pierre-haessig commented:
Hello, I still see the notice "1 change requested" although I did the change in my July commits. Should I press some button?

rgommers commented:
> Hello, I still see the notice "1 change requested" although I did the change in my July commits. Should I press some button?

You just did, by pinging me:) I missed the updates, sorry about that.

rgommers commented:
my comments have been addressed

> Also, Warren disagree with my saying that weibull_min is "the usual Weibull distribution" since in two classical textbooks of extreme value theory it is in fact weibull_max. Still, I think that outside this field there is no ambiguity (and also on Wikipedia page on extreme value theory). I hope that the sentence I added in weibull_min docstring is well balanced.

@WarrenWeckesser could you have a look at this again?

WarrenWeckesser commented on scipy/stats/_continuous_distns.py:
```diff
@@ -1519,20 +1519,39 @@ class exponweib_gen(rv_continuous):
 
-    `exponweib` takes :math:`a` and :math:`c` as shape parameters.
+    * :math:`a` is the exponentiation parameter,
+      with the special case :math:`a=1` corresponding to the usual
```
Because you explicitly refer to `weibull_min` here, you can drop the word "usual" here with no loss of clarity.

rgommers replied:
addressed, so we can merge this. also pushed a rebase

WarrenWeckesser commented:
This looks great.  I made one small suggestion to drop the word "usual" (sorry for being so persnickety).  The way you've written the sentence, the word "usual" doesn't add anything useful to the sentence.  You've explicitly referred to `weibull_min` there, so that sentence is perfectly clear without qualifying the distribution as the "usual" one.

rgommers commented:
failing because of gh-10545:
```
/home/circleci/repo/build/testenv/lib/python3.7/site-packages/scipy/stats/__init__.py:docstring of scipy.stats.exponweib:19: WARNING: py:obj reference target not found: numpy.random.weibull
/home/circleci/repo/build/testenv/lib/python3.7/site-packages/scipy/stats/__init__.py:docstring of scipy.stats.weibull_min:22: WARNING: py:obj reference target not found: numpy.random.weibull
/home/circleci/repo/build/testenv/lib/python3.7/site-packages/scipy/stats/__init__.py:docstring of scipy.stats.weibull_min:35: WARNING: py:obj reference target not found: numpy.random.weibull
```

rgommers commented:
Finally green. The linking to `numpy.random` is going to break again when it's fixed in NumPy, but we'll worry about it then.

Thanks @pierre-haessig! And thanks @WarrenWeckesser for reviewing.

pierre-haessig commented:
Thanks!
